vip-jetpack/connection-pilot/class-jetpack-connection-pilot.php

```diff
@@ -252,16 +269,40 @@ public static function should_run_connection_pilot() {
 	/**
 	 * Checks if a reconnection should be attempted
 	 *
-	 * @param $error \WP_Error Optional error thrown by the connection check
+	 * @param $error \WP_Error|null Optional error thrown by the connection check
+	 *
 	 * @return bool True if a reconnect should be attempted
 	 */
-	public static function should_attempt_reconnection( $error = null ) {
+	public static function should_attempt_reconnection( \WP_Error $error = null ): bool {
+		// The constant is deprecated, but keeping this check for historical reasons
 		if ( defined( 'VIP_JETPACK_CONNECTION_PILOT_SHOULD_RECONNECT' ) ) {
 			return VIP_JETPACK_CONNECTION_PILOT_SHOULD_RECONNECT;
 		}
 
 		return apply_filters( 'vip_jetpack_connection_pilot_should_reconnect', false, $error );
 	}
+
+	/**
+	 * Checks if the site was created less than an hour before execution
+	 *
+	 * @return bool
+	 */
+	public static function is_fresh_subsite(): bool {
+		// Not applicable for single sites
+		if ( ! is_multisite() ) {
+			return false;
+		}
+
+		try {
+			$site_registered = new DateTime( get_site()->registered );
+		} catch ( \Exception $e ) {
+			return false;
+		}
+
+		$now = new DateTime("now");
+		$time_diff = $now->getTimestamp() - $site_registered->getTimestamp();
+		return $time_diff < 3600;
+	}
 }
 
 Connection_Pilot::instance();
```

Comment:
What do you think about booting this all up on the `init` hook instead of right away here? Essentially:

```
add_action( 'init', function() {
  Connection_Pilot::instance();
 }, 25 );
 ```
 
Benefit being customers can use the filters like `vip_jetpack_connection_pilot_should_run` (and constants) in their theme or plugins that way. Otherwise they are pretty restricted to having to do all this before the mu-plugin is loaded, meaning vip-config.php is the only option. In which place you can't really target subsites very easily.

Alternatively, run the `init_actions()` method on the `init` hook instead maybe? Needs compatibility testing specifically with the cron hook, but pretty sure it would be okay.

Reply:
Absolutely. I didn't want to touch this as it wasn't directly related to the PR, but I agree that it adds value. What's more, I've seen this unconditional instantiations of classes in some other places in mu-plugins. I think it's way safer to do it with a hook.

I've pushed a change with your proposed code.